### DISCUSSION TITLE
Schema.org should document usage guidelines for Microdata 'itemid'

### DISCUSSION THREAD

**Discussion by danbri** [schemaorg/schemaorg#2834] [General]
Migrating from https://www.w3.org/2011/webschema/track/issues/6

From http://groups.google.com/group/schemaorg-discussion/browse_thread/thread/62117670d187559e?pli=1

Raised as 

"The concept is defined in 
http://www.whatwg.org/specs/web-apps/current-work/multipage/microdata.html#global-identifiers-for-items 
and means that you can do things like 

<p itemscope itemtype="http://schema.org/Person" 
itemid="http://foolip.org/"><span itemprop="name">Philip</span></p> 

This might seem like an academic questions, but it is what defines if 
validators for schema.org should allow itemid or not. Do any of the 
sponsors do anything with itemid? If not, updating the documentation 
to explicitly say that it is not allowed (and ignored) would be most 
welcome. "

... ensuing discussion made it clear that Schema.org does support itemid. This issue tracks the need to document that on Schema.org.

**Comment by RichardWallis** [comment]
See issue schemaorg/suggestions-questions-brainstorming#7 for the context of the move from the main Schema.org [issue tracker](https://github.com/schemaorg/schemaorg/issues) to this repository.

**Comment by thadguidry** [comment]
@danbri since this is a Discussion item directly and not a GitHub Issue...I guess you are asking for public opinion on it?  But I don't see any underlying question in the subject or the paragraphs?  Was this Discussion item created accidentally and probably best to just be created as a real Issue and delete this Discussion? (I couldn't find Issue #2834).  Also, this might be a weekend bug that happened with GitHub Issues/Discussions when it was created. Dunno.

**Reply by irene459** [reply]
I don't know but what ever bug it was it needs to get lost and stop destroying my life . Am just a simple person that just wants to have a phone like a normal human being and not be all up on GitHub. I don't even know how to communicate on this . The hacker is the one that is all up on this . You tried to unsubscribe but I can't cuz of this dumb ass

**Reply by WeaverStever** [reply]
@irene459 

Hi Irene

Go to your github profile
https://github.com/irene459

In the upper right corner, there should be a drop-down menu.
Select "Settings"
On the Settings page, select "Notifications"
On the Notifications page, you will probably want to uncheck everything.

**Comment by danbri** [comment]
Richard and I were talking about moving the questions-etc repo back into
the main one to use Discussions, it might be a test of that

**Reply by thadguidry** [reply]
Ah, certainly...if it's "Question" oriented, I'd have no problem having some of those here in Discussion where they would fit much more appropriately.

**Reply by thadguidry** [reply]
and by the way... "I'm baccckkkkk!!!!"

**Comment by WeaverStever** [comment]
I've found that the itemid is important. I was doing some testing on the SDTT a few years back, I found that the @id (JSON-LD) and the itemid gives us a handle to the script.

For instance if you are getting a script form an external source that is incomplete, you can provide a better script locally with the same id.

For instance, the first script below fails Rich Snippets Testing Tool because it is missing description and uploadDate. Using the itemid as a handle, I can append the missing properties to the script.

~~~~
<div itemprop="subjectOf" itemscope itemtype="https://schema.org/VideoObject" itemid="http://example.com/#">
        <h2 itemprop:"name">
            Video: <span itemprop="name">Cool artist interview</span>
        </h2>
        <meta itemprop="duration" content="PT7M33S">
        <link itemprop="thumbnailUrl" href="http://example.com/picture.jpg">
        <link itemprop="embedUrl" href="http://example.com/intervideo.html">
        <link itemprop="mainEntityOfPage" href="http://example.com/cool_artist">
</div>


<div itemprop="subjectOf" itemscope itemtype="https://schema.org/VideoObject" itemid="http://example.com/#">
        <meta itemprop="description" content="cool video">
        <meta itemprop="uploadDate" content="Mar 21, 2021"> 
</div>

~~~~

So if the required properties are changed at a later date (a recent example is Events), if you have the id, you can just put another script at the bottom of your page with the missing properties.

**Comment by gkellogg** [comment]
Note that the RDF interpretation of `itemid` is defined in the [Microdata to RDF](https://www.w3.org/TR/microdata-rdf/) note: https://www.w3.org/TR/microdata-rdf/#dfn-itemid. Without using this attribute, it is impossible to control the subject of generated triples along the lines of using `@about/@resource` in RDFa or `@id` in JSON-LD. Removing support for it would severely limit the use of Microdata for reflecting structured data where it is important to define the subject of statements made beyond the page containing it, or a generated blank node.
